File: python/d3.py
from datetime import datetime, timedelta

import pytz

date_string = '2023-01-01T18:00:00-01:00'

# cria um objeto datetime com a data em UTC
utc_date = datetime.fromisoformat(date_string).astimezone(pytz.utc)

# subtrai 3 horas da data em UTC
new_utc_date = utc_date - timedelta(hours=3)

# formata a data e hora no fuso horário UTC
formatted_date_string = new_utc_date.strftime('%Y-%m-%d %H:%M:%S')

print(f"Avise o hotel em {formatted_date_string} de Brasil/SãoPaulo")

File: csharp/d3.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        string dateString = "2023-01-01T18:00:00-01:00";

        // cria um objeto DateTimeOffset com a data em UTC
        DateTimeOffset utcDate = DateTimeOffset.Parse(dateString).ToUniversalTime();

        utcDate = utcDate.AddHours(-3);

        // formata a data e hora no fuso horário UTC
        string formattedDateString = utcDate.ToString("yyyy-MM-dd HH:mm:ss");

        Console.WriteLine($"Avise o hotel em {formattedDateString} de Brasil/SãoPaulo");
    }
}
